tests/foreman/api/test_docker.py

```diff
@@ -725,41 +810,118 @@ def test_promote_multiple_docker_repo_content_view(self):
 
         @Feature: Docker
 
-        @Status: Manual
-
         """
 
-    @stubbed()
+        product = entities.Product(organization=self.org).create()
+        repo = _create_repository(product)
+
+        content_view = entities.ContentView(
+            composite=False,
+            organization=self.org,
+        ).create()
+        content_view.repository = [repo]
+        content_view = content_view.update(['repository'])
+        self.assertEqual(
+            [repo.id], [repo_.id for repo_ in content_view.repository])
+
+        content_view.publish()
+        cvv = content_view.read().version[0]
+        self.assertEqual(len(cvv.read().environment), 1)
+
+        for i in range(1, randint(2, 5)):
+            lce = entities.LifecycleEnvironment(organization=self.org).create()
+            cvv.promote({u'environment_id': lce.id})
+            self.assertEqual(
+                len(content_view.read().version[0].read().environment), i+1)
```

Comment:
Is there any reason to call `content_view.read().version[0].read().environment`? I think you can just call `cvv.read().environment` instead.